From e42e832e40fd1045b14de987d320f313a2ff5cda Mon Sep 17 00:00:00 2001
From: Richard Fitzgerald <rf@opensource.wolfsonmicro.com>
Date: Fri, 18 Apr 2014 10:41:32 +0100
Subject: [PATCH 183/351] ASoC: wm5110: Don't fail open if we can't set wake
 source for IRQ

Failing to set the IRQ as a wake source is not necessarily a fatal
error. Whilst unlikely it is plausible we might not require the IRQ to
wake the AP.

Change-Id: I27f85bad6d762f50cc56d1db8dd19ad09a947c0c
Reported-by: Vitaly Rodionov <vitaly.rodionov@wolfsonmicro.com>
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Richard Fitzgerald <rf@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 sound/soc/codecs/florida.c |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/sound/soc/codecs/florida.c b/sound/soc/codecs/florida.c
index 01273aa..98ba951 100644
--- a/sound/soc/codecs/florida.c
+++ b/sound/soc/codecs/florida.c
@@ -2033,13 +2033,10 @@ static int florida_codec_probe(struct snd_soc_codec *codec)
 	}
 
 	ret = irq_set_irq_wake(arizona->irq, 1);
-	if (ret) {
+	if (ret)
 		dev_err(arizona->dev,
 			"Failed to set DSP IRQ to wake source: %d\n",
 			ret);
-		arizona_free_irq(arizona, ARIZONA_IRQ_DSP_IRQ1, priv);
-		return ret;
-	}
 
 	snd_soc_dapm_enable_pin(&codec->dapm, "DRC2 Signal Activity");
 	ret = regmap_update_bits(arizona->regmap, ARIZONA_IRQ2_STATUS_3_MASK,
-- 
1.7.9.5

